### PR TITLE
Include VSS 3.0 and use it as default

### PR DESCRIPTION
**Pull request by erikbosch**
Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>

### CODE REVIEW

**Comment by erikbosch** [comment]
Draft for discussion/test

File: kuksa-val-server/src/VSSTypeSanitizer.cpp [hunk 1]
```diff
@@ -165,8 +165,23 @@ void VssDatabase::checkAndSanitizeType(jsoncons::json &meta, jsoncons::json &val
         checkBoolType(val);
     }
     else if (dt == "string") { 
+      if (meta.contains("allowed") ) {
+        if (vss_3_0_supported) {
```
Comment (erikbosch) [inline kuksa-val-server/src/VSSTypeSanitizer.cpp:169]: This is one idea how to handle backward compatibility, have constants that specifies versions supported. Fills two purposes - one is to make it easy to remove support for certain versions in "own" forks, another is to make it easy to find version-dependent functionality.

File: data/vss-core/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# KUKSA.val VSS handling
```
Comment (erikbosch) [inline data/vss-core/README.md:1]: An attempt to document what needs to be done when updating VSS version

File: kuksa-val-server/src/VSSTypeSanitizer.cpp [hunk 3]
```diff
@@ -165,8 +165,23 @@ void VssDatabase::checkAndSanitizeType(jsoncons::json &meta, jsoncons::json &val
         checkBoolType(val);
     }
     else if (dt == "string") { 
+      if (meta.contains("allowed") ) {
```
Comment (SebastianSchildt) [inline kuksa-val-server/src/VSSTypeSanitizer.cpp:168]: I would not use those complicated vss_xx_supported constants, I would just silently support both, ie. sth lik

```
if (meta.contains("allowed") {
  checkEnumType(meta["allowed"], val);
}
else if (meta.contains("enum") ) {
checkEnumType(meta["enum"], val);
}
```

**Comment by erikbosch** [comment]
Replaced by https://github.com/eclipse/kuksa.val/pull/331

**Pull request closed without merging**